src/feature/relay/relay_periodic.c

```diff
@@ -202,19 +202,45 @@ reachability_warnings_callback(time_t now, const or_options_t *options)
     /* every 20 minutes, check and complain if necessary */
     const routerinfo_t *me = router_get_my_routerinfo();
     if (me && !router_should_skip_orport_reachability_check(options)) {
-      char *address = tor_dup_ip(me->addr);
-      if (address) {
+      /* We need to warn that one or more of our ORPorts isn't reachable.
+       * Determine which, and give a reasonable warning. */
+      char *address4 = tor_dup_ip(me->addr);
+      char *address6 = tor_addr_to_str_dup(&me->ipv6_addr);
+      bool v4_ok =
+        router_should_skip_orport_reachability_check_family(options,AF_INET);
+      bool v6_ok =
+        router_should_skip_orport_reachability_check_family(options,AF_INET6);
+      if (address4 || address6) {
```

Comment:
So hmm... this entire block can be skipped if `v4_ok` **and** `v6_ok` are `true` thus I would be incline to get those value before and bail if they are true.

If not, then we can start doing the warning dance and only then we can do more expensive address dup(), asprintf and control port message.

Reply:
Should be addressed in b365179ee0061db2981f09025041a1065d0b283c